ckan/config/middleware/flask_app.py

```diff
@@ -517,9 +524,36 @@ def is_blueprint(mm: Any):
 def _register_error_handler(app: CKANApp):
     u'''Register error handler'''
 
-    def error_handler(e: Exception) -> tuple[str, Optional[int]]:
+    def error_handler(e: Exception) -> Union[
+        tuple[str, Optional[int]], Optional[Response]
+    ]:
         debug = config.get_value('debug')
         if isinstance(e, HTTPException):
+            # If the current_user.is_anonymous and the
+            # Exception code is 401(Unauthorized)/403(Forbidden)
+            # force the users to log in before we check for their access.
+            # This logic will be ignored if the ckan.login_required_disabled
+            # is set to True, in your .ini file.
+            if not config.get_value('ckan.login_required.disabled'):
+                # if the url is not local we dont want to redirect the user
+                # instead, we want to show the actual 403(Forbidden)...
+                # same for user.perform_reset if the current_user.is_deleted()
+                # the view returns 403 hence we want to show the exception.
+                endpoints = tuple(
+                    ['util.internal_redirect', 'user.perform_reset']
+                )
+                if request.endpoint not in endpoints:
+                    if current_user.is_anonymous and type(e) in (
+                        Unauthorized, Forbidden
+                    ):
+                        login_url = h.url_for('user.login', qualified=True)
+                        next_url = request.url
+                        redirect_url = h.make_login_url(
+                            login_url, next_url=next_url
+                        )
+                        h.flash('Please log in to access this page.')
```

Comment:
I'm not seeing any flash messages when being redirected. Also this message needs to be translated:
```suggestion
                        h.flash_error(_('Please log in to access this page'))
```

Reply:
<img width="1245" alt="image" src="https://user-images.githubusercontent.com/72216462/175954062-c157a05f-c9bf-4550-9a91-38b7fb43a6f1.png">

I m actually, seeing two error messages,  but you are right, they are not constantly showing, sometimes they pop up sometimes they don't, and I m not sure about the real reason behind this.

@amercader